commands/command_fetch.go

```diff
@@ -278,6 +278,9 @@ func fetchAndReportToChan(pointers []*lfs.WrappedPointer, include, exclude []str
 		// This avoids previous case of over-reporting a requirement for files we already have
 		// which would only be skipped by PointerSmudgeObject later
 		passFilter := lfs.FilenamePassesIncludeExcludeFilter(p.Name, include, exclude)
+
+		lfs.IfNoLocalObjExistsLinkOrCopyFromReferenceMedia(p.Oid, p.Size)
```

Comment:
This should be added to the appropriate place in `command_smudge.go` too. Possibly [here](https://github.com/github/git-lfs/blob/70e5df64acdcb4f86077e1399b0df0925435ed9b/commands/command_smudge.go#L39).